macros/edr/dbt_artifacts/upload_dbt_columns.sql

```diff
@@ -0,0 +1,67 @@
+{%- macro upload_dbt_columns(should_commit=false, metadata_hashes=none) -%}
+    {% set relation = elementary.get_elementary_relation('dbt_columns') %}
+    {% if execute and relation %}
+        {% set tables = graph.nodes.values() | list + graph.sources.values() | list %}
+        {% do elementary.upload_artifacts_to_table(relation, tables, elementary.flatten_table_columns, should_commit=should_commit, metadata_hashes=metadata_hashes) %}
+    {%- endif -%}
+    {{- return('') -}}
+{%- endmacro -%}
+
+{% macro get_dbt_columns_empty_table_query() %}
+    {% set columns = [
+        ('unique_id', 'string'),
+        ('parent_unique_id', 'string'),
+        ('name', 'string'),
+        ('data_type', 'string'),
+        ('tags', 'long_string'),
+        ('meta', 'long_string'),
+        ('database_name', 'string'),
+        ('schema_name', 'string'),
+        ('table_name', 'string'),
+        ('description', 'long_string'),
+        ('generated_at', 'string'),
+        ('metadata_hash', 'string'),
+    ] %}
+    {% set dbt_columns_empty_table_query = elementary.empty_table(columns) %}
+    {{ return(dbt_columns_empty_table_query) }}
+{% endmacro %}
+
+{% macro flatten_table_columns(table_node) %}
+    {% set column_nodes = table_node.get("columns") %}
+    {% if not column_nodes %}
+        {% do return(none) %}
+    {% endif %}
+
+    {% set flattened_columns = [] %}
+    {% for column_node in column_nodes.values() %}
+        {% do flattened_columns.append(elementary.flatten_column(table_node, column_node)) %}
+    {% endfor %}
+    {% do return(flattened_columns) %}
+{% endmacro %}
+
+{% macro flatten_column(table_node, column_node) %}
+    {% set config_dict = elementary.safe_get_with_default(column_node, 'config', {}) %}
+    {% set config_meta_dict = elementary.safe_get_with_default(config_dict, 'meta', {}) %}
+    {% set meta_dict = elementary.safe_get_with_default(column_node, 'meta', {}) %}
+    {% do meta_dict.update(config_meta_dict) %}
+    {% set config_tags = elementary.safe_get_with_default(config_dict, 'tags', []) %}
+    {% set global_tags = elementary.safe_get_with_default(column_node, 'tags', []) %}
+    {% set meta_tags = elementary.safe_get_with_default(meta_dict, 'tags', []) %}
+    {% set tags = elementary.union_lists(config_tags, global_tags) %}
+    {% set tags = elementary.union_lists(tags, meta_tags) %}
+    {% set flatten_column_metadata_dict = {
```

Comment:
Should we add `resource_type`?

Reply:
Can help with joins, no?
Sounds like a good idea